File: api/src/main/java/br/com/megaloja/dtos/OrderResponse.java
package br.com.megaloja.dtos;

import br.com.megaloja.models.enums.OrderStatus;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@JsonPropertyOrder({"id", "customerId", "customerName", "storeId", "storeName", "status", "pickupCode", "totalAmount", "pickupDeadline", "createdAt", "updatedAt", "items"})
public record OrderResponse(
        Long id,
        Long customerId,
        String customerName,
        Long storeId,
        String storeName,
        OrderStatus status,
        String pickupCode,
        BigDecimal totalAmount,
        LocalDateTime pickupDeadline,
        LocalDateTime createdAt,
        LocalDateTime updatedAt,
        List<OrderItemResponse> items
) {
}

File: web/src/app/core/models/order-response.ts
export interface OrderResponse {
  id: number;
  customerId: number;
  customerName: string;
  storeId: number;
  storeName: string;
  status: string;
  pickupCode: string;
  totalAmount: number;
  pickupDeadline: string;
  createdAt: string;
  updatedAt: string;
  items: unknown[];
}
